### PR TITLE
Replace validationState with isInvalid

### PR DESCRIPTION
**Pull request by devongovett**
Closes https://github.com/orgs/adobe/projects/19/views/32?pane=issue&itemId=36181573

This replaces the `validationState` prop with `isInvalid` in all of our APIs. `validationState` is a concept from Spectrum that has leaked into our other APIs. It was intended to cover the green checkmark use case in text fields and other input-like components (e.g. ComboBox and DateField). Other components such as RadioGroup and Checkbox only support `validationState="invalid"` and not `validationState="valid"`, so this prop doesn't make much sense. Outside of Spectrum, this concept of a valid state doesn't exist at all making for a more confusing API.

After this PR, all components in RAC as well as React Aria hooks have moved to `isInvalid` instead. Spectrum retains `validationState` only for input field like components, but moves to `isInvalid` for others such as Checkbox and Picker. `validationState` still works in React Aria hooks and React Spectrum for backward compatibility, but is deprecated. The prop is completely replaced by `isInvalid` in React Aria Components, which is still in alpha.

React Aria Components also moves from using `data-validation-state` to consistently using `data-invalid` everywhere.

### CODE REVIEW

**Review by reidbarber — DISMISSED** [review]
LGTM, lets get this in for a testing session

File: packages/@react-spectrum/calendar/src/CalendarBase.tsx [hunk 1]
```diff
@@ -136,12 +136,12 @@ export function CalendarBase<T extends CalendarState | RangeCalendarState>(props
           onClick={() => state.focusNextPage()}
           tabIndex={-1} />
       </VisuallyHidden>
-      {state.validationState === 'invalid' &&
+      {state.isInvalid &&
```
Comment (LFDanLu) [inline packages/@react-spectrum/calendar/src/CalendarBase.tsx:139]:
```suggestion
      {state.isValueInvalid &&
```
Noticed some breaks in chromatic: https://www.chromatic.com/test?appId=5f0dd5ad2b5fc10022a2e320&id=64e92f45a003125f7a68145c. Running it again locally with the above change, will push if it runs clean

**Review by LFDanLu — CHANGES_REQUESTED** [review]

**Review by LFDanLu — APPROVED** [review]
Did a brief sweep of the docs/stories and verified that RSP tests ran clean even after reverting them to main so seems like the back compatibility is working.

File: packages/@react-stately/radio/src/useRadioGroupState.ts [hunk 2]
```diff
@@ -76,6 +82,7 @@ export function useRadioGroupState(props: RadioGroupProps): RadioGroupState  {
     isDisabled: props.isDisabled || false,
     isReadOnly: props.isReadOnly || false,
     isRequired: props.isRequired || false,
-    validationState: props.validationState || null
+    validationState: props.validationState || null,
+    isInvalid: props.isInvalid || props.validationState === 'invalid'
```
Comment (snowystinger) [inline packages/@react-stately/radio/src/useRadioGroupState.ts:86]: these should probably all be like this?
```suggestion
    isInvalid: props.isInvalid ?? props.validationState === 'invalid'
```

File: packages/@react-aria/radio/src/useRadioGroup.ts [hunk 3]
```diff
@@ -127,7 +126,7 @@ export function useRadioGroup(props: AriaRadioGroupProps, state: RadioGroupState
       // https://www.w3.org/TR/wai-aria-1.2/#radiogroup
       role: 'radiogroup',
       onKeyDown,
-      'aria-invalid': validationState === 'invalid' || undefined,
+      'aria-invalid': state.isInvalid || undefined,
```
Comment (snowystinger) [inline packages/@react-aria/radio/src/useRadioGroup.ts:129]: to stay backwards compatible, this hook should still include validationState no?

File: packages/@react-aria/radio/docs/useRadioGroup.mdx [hunk 4]
```diff
@@ -113,7 +113,7 @@ function RadioGroup(props) {
         {children}
       </RadioContext.Provider>
       {description && <div {...descriptionProps} style={{fontSize: 12}}>{description}</div>}
-      {errorMessage && validationState === 'invalid' &&
+      {errorMessage && state.isInvalid &&
```
Comment (snowystinger) [inline packages/@react-aria/radio/docs/useRadioGroup.mdx:116]: need to keep validationState i think for backwards compact?

File: packages/@react-spectrum/calendar/stories/RangeCalendar.stories.tsx [hunk 5]
```diff
@@ -74,9 +74,8 @@ export default {
       control: 'select',
       options: [null, 'single', 'visible']
     },
-    validationState: {
-      control: 'select',
-      options: [null, 'valid', 'invalid']
+    isInvalid: {
```
Comment (snowystinger) [inline packages/@react-spectrum/calendar/stories/RangeCalendar.stories.tsx:77]: these (and a few others in this PR) are RSP components, they should still be validationState? or is the idea that these are components that don't have a valid validation state, only invalid?

**Review by snowystinger — COMMENTED** [review]
A lot of this hinges on people not having written their own stately hooks. I wish we had some data on that. We could guard against it a little more with
`state.isValid ?? state.validationState ...` like we do in stately itself

File: packages/@react-spectrum/calendar/stories/RangeCalendar.stories.tsx (hunk 5 above)
Comment (devongovett) [reply packages/@react-spectrum/calendar/stories/RangeCalendar.stories.tsx:77]: from pr description:

> Spectrum retains validationState only for input field like components, but moves to isInvalid for others such as Checkbox and Picker

File: packages/@react-aria/radio/docs/useRadioGroup.mdx (hunk 4 above)
Comment (devongovett) [reply packages/@react-aria/radio/docs/useRadioGroup.mdx:116]: `state.isInvalid` also takes `validationState` into account

File: packages/@react-stately/radio/src/useRadioGroupState.ts (hunk 2 above)
Comment (devongovett) [reply packages/@react-stately/radio/src/useRadioGroupState.ts:86]: hmm not sure we should treat `false` and `undefined` differently. I think it's ok if it's invalid if either prop says so.

**Comment by devongovett** [comment]
> A lot of this hinges on people not having written their own stately hooks. I wish we had some data on that.

I think TypeScript should guard against that? The types won't be compatible if they are missing an `isInvalid` property for example.

**Review by LFDanLu — DISMISSED** [review]

**Review by reidbarber — DISMISSED** [review]

**Review by LFDanLu — APPROVED** [review]

**Pull request merged**